src/ball_tree.jl

```diff
@@ -15,87 +15,109 @@ end
 # When we create the bounding spheres we need some temporary arrays.
 # We create a type to hold them to not allocate these arrays at every
 # function call and to reduce the number of parameters in the tree builder.
-immutable ArrayBuffers{T <: AbstractFloat}
-    left::Vector{T}
-    right::Vector{T}
-    v12::Vector{T}
-    zerobuf::Vector{T}
+immutable ArrayBuffers{N, T <: AbstractFloat}
+    left::MVector{N, T}
```

Comment:
These are only used in the creation of a tree and there are almost no computations done on them. They are also required to be mutable. The actual `HyperSpheres` that are stored in the tree are using `SVectors` as you can see for compact data storage.